packages/api/src/controllers/managementPlanController.js

```diff
@@ -14,23 +14,106 @@
  */
 
 import ManagementPlanModel from '../models/managementPlanModel.js';
+import ManagementPlanGroup from '../models/managementPlanGroupModel.js';
 import CropManagementPlanModel from '../models/cropManagementPlanModel.js';
 import ManagementTasksModel from '../models/managementTasksModel.js';
 import TaskModel from '../models/taskModel.js';
 import TaskTypeModel from '../models/taskTypeModel.js';
 import FieldWorkTypeModel from '../models/fieldWorkTypeModel.js';
 import TransplantTaskModel from '../models/transplantTaskModel.js';
 import PlantTaskModel from '../models/plantTaskModel.js';
-import { raw } from 'objection';
-import lodash from 'lodash';
+import UserFarmModel from '../models/userFarmModel.js';
+import objection, { raw } from 'objection';
+import _pick from 'lodash/pick.js';
 import { sendTaskNotification, TaskNotificationTypes } from './taskController.js';
+import {
+  getDatesFromManagementPlanGraph,
+  getManagementPlanGroupTemplateGraph,
+  getFormattedManagementPlanData,
+} from '../util/copyCropPlan.js';
+import { getSortedDates } from '../util/util.js';
+const { transaction, Model } = objection;
 
 const managementPlanController = {
+  repeatManagementPlan() {
+    return async (req, res) => {
+      const trx = await transaction.start(Model.knex());
+      const { startDates, management_plan_id, repeatDetails } = req.body;
```

Comment:
What do you think `repeatDeails` should look like? The user input will be:
```
{
    crop_plan_name: 'Repetitions of Apricot plan template',
    plan_start_date: '2023-07-25',
    repeat_frequency: 1,
    repeat_interval: { value: 'year', label: 'year(s)' },
    days_of_week: ['Tuesday'],
    finish: 'after',
    after_occurrences: '5',
  }
```
The only thing that matters for now is if I add `planName` or change `crop_plan_name` to `planName`.

Reply:
Whatever works for you ! I have no preference.  I think I was working off of that last slack conversation about the shape with Joyce.

It probably make sense for me to change so your names stay consistent snake case.

Reply:
I think it's more natural to leave the user input as is, too.
Could you use `crop_plan_name`?

Reply:
got it, will do!